lib/main.dart

```diff
@@ -9,8 +10,13 @@ class MyApp extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final MovieRepository repository = MovieRepository();
```

Comment:
This should be `final IMovieRepository repository = MovieRepository();`!
The presentation layer uses abstractions from the domain layer

Reply:
fixed! Also moved the instantiation of the MovieRepository to the main function, so MyApp widget no longer depends on that implementation. This means I can pass a different repo implementation to MyApp in yest/widget.test.dart (I pass the same implementation for now, though)